superset-frontend/src/components/Menu/SubMenu.tsx

```diff
@@ -66,6 +81,10 @@ export interface SubMenuProps {
   name: string;
   children?: MenuChild[];
   activeChild?: MenuChild['name'];
+  /* If usesRouter is true, a react-router <Link> component will be used instead of href.
+   *  ONLY set usesRouter to true if SubMenu is wrapped in a react-router <Router>;
+   *  otherwise, a 'You should not use <Link> outside a <Router>' error will be thrown */
+  usesRouter?: boolean;
```

Comment:
What do you think about checking if `props.history` exists to determine if we have access to react-router?

Reply:
This is my first time using `react-router`, are you talking about the `useHistory` hook?

Reply:
Sorry, I assumed that we use `withRouter` for pages that use react-router - that would provide `history` object to props. However, do we need to explicitly say that `DatasetList` and `DatasourceList` use router? I think we only use them inside `<Router>` in `views/App` file, so maybe we could drop `usesRouter` prop for them? Sorry if I confuse something, I'm new to the project 🙂

Reply:
Did some digging, looks like `useHistory` is just another way to access the router's history object, so I've updated the `SubMenu` to check for that! This will allow implicit Router checking.